assets/src/infrastructure/ui/display/TimezoneTimeInfo/index.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import {
+	IconButton,
+	Popover,
+	PopoverTrigger,
+	PopoverContent,
+	PopoverHeader,
+	PopoverBody,
+	PopoverArrow,
+	PopoverCloseButton,
+} from '@chakra-ui/core';
+import { __ } from '@wordpress/i18n';
+import { GlobalOutlined } from '@ant-design/icons';
+
+import { useTimeZoneTime } from '@appServices/hooks';
+import './style.scss';
+
+export interface Props {
+	date: Date;
+}
+
+const TimezoneTimeInfo: React.FC<Props> = ({ date }) => {
+	const { formatDateForSite, formatDateForUser, formatUtcDateForSite } = useTimeZoneTime();
+	const content = (
+		<div>
+			<div className={'ee-focus-priority-8'}>
+				<strong>{__('Your Local Time Zone')}</strong>
+			</div>
+			<div className={'ee-focus-priority-6'}>{formatDateForUser(date)}</div>
+			<br />
+			<div className={'ee-focus-priority-8'}>
+				<strong>{__("The Website's Time Zone")}</strong>
+			</div>
+			<div className={'ee-focus-priority-6'}>{formatDateForSite(date)}</div>
+			<br />
+			<div className={'ee-focus-priority-8'}>
+				<strong>{__('UTC (Greenwich Mean Time)')}</strong>
+			</div>
+			<div className={'ee-focus-priority-6'}>{formatUtcDateForSite(date)}</div>
+		</div>
+	);
+	const label = __('click for timezone\ninformation');
+
+	return (
+		<div className='ee-timezone-info'>
+			<Popover>
+				<PopoverTrigger>
+					<IconButton aria-label={label} icon={GlobalOutlined} variant='unstyled' />
+				</PopoverTrigger>
+				<PopoverContent zIndex={4}>
+					<PopoverArrow />
+					<PopoverCloseButton className='ee-timezone-info__close-button' color='var(--ee-color-black)' />
+					<PopoverHeader color='var(--ee-color-black)'>{__('This Date Converted To:')}</PopoverHeader>
+					<PopoverBody>{content}</PopoverBody>
+				</PopoverContent>
+			</Popover>
```

Comment:
**THIS** composition of things from Chakra would be appropriate as some kind of "popover" adapter within the infrastructure layer as long as it does NOT set classes like "ee-timezone-info" or apply labels like "This Date Converted To" because those are specific to a single use case.

Remember the little description I posted yesterday about how layers are setup:

```
third party code -> infrastructure -> application -> domain
```

which can also be worded as:

```
external services & objects -> adapters -> framework -> use cases
```

There's two main ideas that are really important to understand about how they interact:

1. the flow of dependencies (which in the simplest form can be thought of as imports) goes from left to right (also described as flowing from the outer layers to the inner layers, because the domain layer is essentially wrapped bv the other layers: like in this [image](https://camo.githubusercontent.com/081561c762f6fcb27bccd8c647e7ef0709c7d21f/68747470733a2f2f63646e2e7261776769742e636f6d2f6a6b70686c2f636c6561722d6172636869746563747572652f62316439643366322f696d616765732f636c6561722d6172636869746563747572652d646f6d61696e2d6170706c69636174696f6e2d636c69656e742d74696572732e737667) or this more [complex image](https://herbertograca.files.wordpress.com/2018/11/100-explicit-architecture-svg.png)):

    - third party code obviously can not depend on any of our code 

    - adapters should not depend on anything other than third party code as they are simply a way of preparing external logic for our use within our application (as well as keeping things decoupled)

    - the application layer should consist primarily of its own logic but can utilize adapters to compose services and elements that can then be used within all of our specific use cases

    - the domain layer is all of our specific use cases and will primarily depend on the application layer but can also directly import adapters from the infrastructure layer (although it's better to utilize logic from the app layer). Because this layer represents the specific use cases, none of its logic can be utilized in any of the other layers. For example, the current use case we are working on is the event editor in the admin. If any of that specific logic was utilized within a component in the application (or infrastructure) layer, then that component would not be usable in any other use cases (or more accurately, it would indicate some logic that is not specific to the event editor and should therefore be relocated to another layer, or at least into the "shared" folder within the domain layer). You wouldn't want content that requires user authentication popping up in non-authed use cases.

2. logic becomes more and more specific and specialized as you move from the outer layers to the inner layers

    - considering the outermost layer is comprised of third party code / external services, it's not hard to see how that would be the most generalized type of code. For example, a database service or a UI library like Chakra could be used to build just about anything as they do not define any specific use case.

    - the adapters in the infrastructure layer are a an intermediary step that prepare those completely generic external services and objects for use within our own application. So you can see that things suddenly got quite a bit more specialized because the logic in that layer is only usable by our application.

    - the application layer takes this a step further by building up services and objects that are more and more specialized (like the Entity List component) but still not specific to one specific use case.

    - the domain layer is of course comprised of all of the specific use cases, although we only have one at the moment (the admin event editor), and consists of the most specialized logic of all of the layers. Anything within a specific use case should be unusable and/or unwanted in any other use case or layer. If that is not the case, then it indicates that the logic should be extracted into a location with greater scope.

Let me know if you have any questions about this. Maybe we can call discuss this at the next dev meeting so that everyone is on the same page (I should have done this already so my apologies for any confusion)

Reply:
Yes, it clicked, thx.
I was confused/ biased by how `antd` components were implemented before this approach.
Some of `antd` components being wrapped inside ours, but other being used directly.